client/controllers/downloadCSVModal.coffee

```diff
@@ -1,10 +1,29 @@
+Template.downloadCSVModal.onCreated ->
+  @prepared = new ReactiveVar(null)
+  @preparing = new ReactiveVar(false)
+
+  @autorun =>
+    if @prepared.get()
+      @preparing.set(false)
+
+Template.downloadCSVModal.onRendered ->
+  @data.rendered()
+
 Template.downloadCSVModal.helpers
   getField: (row, field)->
     row[field]
 
+  preparing: ->
+    Template.instance().preparing.get()
+
 Template.downloadCSVModal.events
   'click .download-csv': (event, instance) ->
+    instance.prepared.set(null)
+    instance.preparing.set(true)
     fileType = $(event.currentTarget).attr('data-type')
     table = instance.$('table')
     if table.length
-      table.tableExport(type: fileType)
+      # Delay export so UI can respond to change in reactiveVars
+      setTimeout ->
```

Comment:
@nathanathan Is there a better way to handle this? I suppose the logic in the event runs immediately after the reactiveVar is set and blocks the reactivity (the helper and subsequently the UI change)?